metricsaggregator/src/main/java/io/dockstore/metricsaggregator/client/cli/TerraMetricsSubmitter.java

```diff
@@ -0,0 +1,572 @@
+import io.dockstore.metricsaggregator.client.cli.CommandLineArgs.SubmitTerraMetrics.TerraMetricsCsvHeaders;
+import io.dockstore.openapi.client.ApiClient;
+import io.dockstore.openapi.client.ApiException;
+import io.dockstore.openapi.client.api.ExtendedGa4GhApi;
+import io.dockstore.openapi.client.api.WorkflowsApi;
+import io.dockstore.openapi.client.model.ExecutionsRequestBody;
+import io.dockstore.openapi.client.model.RunExecution;
+import io.dockstore.openapi.client.model.RunExecution.ExecutionStatusEnum;
+import io.dockstore.openapi.client.model.Workflow.DescriptorTypeEnum;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TerraMetricsSubmitter {
+    private static final Logger LOG = LoggerFactory.getLogger(MetricsAggregatorClient.class);
+    private static final int MAX_NUMBER_OF_MICRO_SECONDS = 9;
+    // Example workflow_start: 2022-07-15 15:37:06.450000 UTC
+    private static final DateTimeFormatter WORKFLOW_START_FORMAT = new DateTimeFormatterBuilder()
+            .appendPattern("yyyy-MM-dd HH:mm:ss")
+            .appendFraction(ChronoField.MICRO_OF_SECOND, 0, MAX_NUMBER_OF_MICRO_SECONDS, true) // There are varying widths of micro seconds
+            .optionalStart() // Start optional time zone pattern. Time zone is not always included
+            .appendPattern(" ")
+            .appendZoneId()
+            .optionalEnd() // End optional time zone pattern
+            .toFormatter();
+    private final MetricsAggregatorConfig config;
+    private final SubmitTerraMetrics submitTerraMetricsCommand;
+    private final AtomicInteger numberOfExecutionsProcessed = new AtomicInteger(0);
+    private final AtomicInteger numberOfExecutionsSubmitted = new AtomicInteger(0);
+    private final AtomicInteger numberOfExecutionsSkipped = new AtomicInteger(0);
+
+    // Keep track of sourceUrls that are found to be ambiguous
+    private final Map<String, String> skippedSourceUrlsToReason = new ConcurrentHashMap<>();
```

Comment:
Suggest defining this and the next 2 maps as `ConcurrentMap`s -- I was looking at the code below for any threading issues, and initially thought there might be an issue until I realized they are concurrent.

I know this wouldn't change the behavior, which is already correct; it's for readability.